core-v1/src/main/scala/engine/StandardEngine.scala

```diff
@@ -98,6 +101,36 @@ object StandardEngine {
       ExprResult.Combine(expr, finalState, left, right)
     }
 
+    //    type SimpleFn[-I, +O] = ExprState[Any, I] => ExprResult[Any, Any, O, OP]
+    //    type StateFn[-I, +O] = ExprState[Any, I] => ExprState[PO, O]
+    //    implicit val arrow: Arrow[StateFn] = new Arrow[StateFn] {
+    //      override def lift[A, B](f: A => B): StateFn[A, B] = _.mapOutput(f)
+    //      override def compose[A, B, C](f: StateFn[B, C], g: StateFn[A, B]): StateFn[A, C] = f.compose(g)
+    //      override def first[A, B, C](fa: StateFn[A, B]): StateFn[(A, C), (B, C)] = { s =>
+    //        val (a, c) = s.output
+    //        val b = fa(s.withOutput(a))
+    //        b.mapOutput((_, c))
+    //      }
+    //    }
+
+    //    object asSimpleFn
+    //      extends Expr.ProxyVisitor[VisitorParam[PO, OP]#Fn, StateFn, OP](this) {
+    //      override protected def proxy[I, O](underlying: => PO <:< I => ExprResult[PO, I, O, OP]): StateFn[I, O] = { _ =>
+    //        val result = underlying(implicitly[PO <:< I])
+    //        result.state
+    //      }
+    //    }
+
+    // TODO: Refactor this engine to better support NonEmptyExprHList
+    override def visitConcatToHList[I, F[+_], WL <: HList : OP, UL <: HList](
+      expr: Expr.ConcatToHList[I, F, WL, UL, OP],
+    ): PO <:< I => ExprResult[PO, I, WL, OP] = { implicit evPOisI =>
+      //      val compute = expr.exprHList.toHListWrappedWith(ToHListWrapped.proxy(asSimpleFn))
+      //      val result = compute(evPOisI(state.output))
+      //      result
+      ???
+    }
```

Comment:
I don't really need the "standard" expression interpreter, so I can punt on this problem for now.